src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.h

```diff
@@ -120,22 +120,9 @@ class MTROperationalCredentialsDelegate : public chip::Controller::OperationalCr
         chip::FabricId fabricId, const chip::CATValues & cats, const chip::Crypto::P256PublicKey & pubkey,
         chip::MutableByteSpan & noc);
 
-    /**
-     * When a NOCChainIssuer is set, then onNOCChainGenerationNeeded will be called when the NOC CSR needs to be
-     * signed. This allows for custom credentials issuer implementations, for example, when a proprietary cloud API will perform the
-     * CSR signing. The commissioning workflow will stop upon the onNOCChainGenerationNeeded callback and resume once
-     * onNOCChainGenerationComplete is called.
-     *
-     * Caller must pass a non-nil value for the rootCertificate, intermediateCertificate, operationalCertificate
-     * If ipk and adminSubject are non nil, then they will be used in the AddNOC command sent to the commissionee. If they are not
-     * populated, then the values provided in the MTRDeviceController initialization will be used.
-     */
-    void onNOCChainGenerationComplete(NSData * operationalCertificate, NSData * intermediateCertificate, NSData * rootCertificate,
-        NSData * _Nullable ipk, NSNumber * _Nullable adminSubject, NSError * __autoreleasing * error);
-
-    void setNSError(CHIP_ERROR err, NSError * __autoreleasing * outError);
-
-    CHIP_ERROR CallbackGenerateNOCChain(const chip::ByteSpan & csrElements, const chip::ByteSpan & csrNonce,
+    void ExternalNOCChainGenerated(MTROperationalCertificateInfo * _Nullable info, NSError * _Nullable error);
```

Comment:
addition of documentation of here would be useful

Reply:
Can do.